### DISCUSSION TITLE
.NET Framework Runtime and Builder Docker repos have merged

### DISCUSSION THREAD

**Discussion by MichaelSimons** [microsoft/dotnet-framework-docker#915] [Announcements]
# .NET Framework Runtime and Builder Docker repos have merged

The [microsoft/dotnet-framework-build](https://hub.docker.com/r/microsoft/dotnet-framework-build/) Docker Hub repo has been merged into the [microsoft/dotnet-framework](https://hub.docker.com/r/microsoft/dotnet-framework/) Docker Hub repo.  All new tags going forward will only be pushed to [microsoft/dotnet-framework](https://hub.docker.com/r/microsoft/dotnet-framework/).  The images in the [microsoft/dotnet-framework-build](https://hub.docker.com/r/microsoft/dotnet-framework-build/) Docker Hub repo will be retained and serviced accordingly as to not affect existing usage.

## Details

The [microsoft/dotnet-framework-build](https://hub.docker.com/r/microsoft/dotnet-framework-build/) Docker Hub repo has been merged into the [microsoft/dotnet-framework](https://hub.docker.com/r/microsoft/dotnet-framework/) Docker Hub repo.

* https://github.com/Microsoft/dotnet-framework-docker/pull/123
* https://github.com/Microsoft/dotnet-framework-docker/issues/78

In order to differentiate the runtime and build images, `runtime` and `sdk` distinguishing identifiers have been added to the tags.  For example, `4.7.2-runtime` and `4.7.2-sdk`.

All new tags going forward will only be pushed to [microsoft/dotnet-framework](https://hub.docker.com/r/microsoft/dotnet-framework/).   For example, the build images for 4.7.2 which was released today, only exist in [microsoft/dotnet-framework](https://hub.docker.com/r/microsoft/dotnet-framework/) repo.

The existing tags in [microsoft/dotnet-framework-build](https://hub.docker.com/r/microsoft/dotnet-framework-build/) will remain and be serviced accordingly.  For example, the stable tags such as `4.7.1-windowsservercore-1709` will continue to be patched on a monthly basis.

We recommend that as you create and maintain your code, you move to use the new `runtime` and `sdk` images from the [microsoft/dotnet-framework](https://hub.docker.com/r/microsoft/dotnet-framework/) repo.

The [microsoft/dotnet-framework-build](https://hub.docker.com/r/microsoft/dotnet-framework-build/) Docker Hub repo will be deleted on 1/1/2019.

**Comment by robrich** [comment]
Is this only for `microsoft/dotnet` docker images or will `microsoft/aspnetcore` images follow this new convention too?

**Comment by MichaelSimons** [comment]
@robrich - this only applies to the `microsoft/dotnet-framework` images - i.e. the full .NET Framework.  The `microsoft/dotnet` images already followed this convention.  `microsoft/aspnetcore` is being transitioned in a different sorts in 2.1.  See https://github.com/aspnet/aspnet-docker/issues/403 for details.

**Comment by MichaelSimons** [comment]
Closing issue but feel free to continue the conversation.

**Comment by tstrass** [comment]
What are the images tagged without `-sdk` or `-runtime`, e.g. `dotnet-framework:4.7.1`?

**Comment by MichaelSimons** [comment]
They are equivalent to the `-runtime` images.  We recommend moving off of these tags onto the `-runtime` tags.  They are no longer being documented.

**Comment by rahulrai-in** [comment]
What is the recommendation for building .net Core SPA images? dotnet build restores the node and nuget packages and therefore requireds both npm and dotnet to be present in the image. aspnetcore-build had both the dependencies available.
